### ISSUE TITLE
Add Event Log summary to Task/Dag Overview tabs

### ISSUE THREAD

**Issue by fritz-astronomer** [apache/airflow#41312]
### Description

@o-nikolas @vincbeck 

During the town hall meeting - it was brought up that important scheduler events in the Event Tab may not be easily visible or discoverable, with folks used to looking at the task logs. 

One idea, was to have a count of Event Logs associated with a particular task run - e.g. `[Event Logs (5)]`

### Related issues
https://github.com/apache/airflow/pull/40967
https://github.com/apache/airflow/pull/40867

### Are you willing to submit a PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

**Comment by bbovenzi** [comment]
Instead of the count in the tab, we can do this on the dag/task overview pages. Although event/audit logs can still be inconsistent on what is recorded.

**Comment by naseemshah** [comment]
i'd like to work on this, and I want to float a slightly richer UX.

How about something like "Smart Logs/ Unified Logs" and a filter menu for log type?

```
[2026-07-14 10:02:01] INFO - Starting attempt 1 of 3
[2026-07-14 10:02:05] INFO - Executing <Task(PythonOperator): extract>
  ─── ⚑ 1 event · 10:02:05 · task started (scheduler) ───────────────
[2026-07-14 10:04:31] INFO - Processing batch 3/12...
  ─── ⚑ 2 events · 10:05:10 · heartbeat timeout, retry scheduled ────
[2026-07-14 10:05:12] INFO - Received SIGTERM. Terminating...
```

Something like this but with interactive rich log lines?

Markers are collapsed one-liners, expandable to full event details. Since the Logs tab already has source and log-level filters, audit events could simply appear as an additional source there — off by default or toggleable, so nothing changes for users who don't want it.

**Comment by bbovenzi** [comment]
Yeah that sounds interesting. Feel free to comment screenshots/videos of a demo here first for us to experiment with the UX before you work on a full pull request
